src/result/result.factory.ts

```diff
@@ -0,0 +1,38 @@
+import { patchPrototype, instantiateWithFactory } from '@utils/factory';
+import { Result } from '@generated/result.generated';
+import { SyncResult } from '@generated/syncResult.generated';
+import { AsyncResult } from '@generated/asyncResult.generated';
+import { AbortedClass } from './aborted';
+import { ErrorClass } from './error';
+import { OkClass } from './ok';
+import { PendingClass } from './pending';
+
+
+/**
+ * Patch the function prototypes to inherit from their respective
+ * Class prototypes
+ */
+function patchFactoryFunctions() {
```

Comment:
Pretty minor, but why not just remove the function and do the patching directly?

Reply:
> Are Result objects still constructible without the `new` keyword now that you use proper Typescript constructors?

I am sure exactly what you mean? Do mean this: 
```ts
const res = new Result('some value');
```
The code above won't work, given that the Result class is marked as `abstract` (same story for optional). But i can change it if necessary. (I can make the Result class non-abstract and add a constructor function)

> Pretty minor, but why not just remove the function and do the patching directly?
To reduce duplication mostly,  i think it is clearer and clearer that way too.

Reply:
My constructor question is more about instantiating Ok or Error directly. Can you still do `return Result.Ok(x)`?

For the patching function, apologies, I hadn't realize the function was called multiple times. I thought it was called only once.

Reply:
Ahh i see, as of right now the `Result.Ok(...)` only works without the `new` keyword, but that is something i can change *easily* (should be at least) .

Currently the `ok` Type function looks like this:
```ts
export type createOkType =  <T = any>(value?: T): SyncResult<T>;
```

but i can change it to:
```ts
export type createOkType = {
    <T = any>(value?: T): SyncResult<T>;
    new <T = any>(value?: T): SyncResult<T>;
};
```

Which would allow both the `Result.Ok(x)` and `new Result.Ok(x)`